res/controllers/Numark-Mixtrack-Pro-FX-scripts.js

```diff
@@ -0,0 +1,926 @@
+        group: "[Library]",
+        inKey: "GoToItem"
+    });
+
+    this.setBeatgrid = new components.Button({
+        group: "[Channel1]",
+        key: "beats_translate_curpos",
+        midi: [0x88, 0x01],
+        off: 0x01,
+        shift: function() {
+            this.group = "[Channel2]";
+        },
+        unshift: function() {
+            this.group = "[Channel1]";
+        }
+    });
+};
+
+MixtrackProFX.Browse.prototype = new components.ComponentContainer();
+
+MixtrackProFX.HeadGain = function() {
+    components.Pot.call(this);
+};
+
+MixtrackProFX.HeadGain.prototype = new components.Pot({
+    group: "[Master]",
+    inKey: "headGain"
+});
+
+MixtrackProFX.vuCallback = function(value, group) {
+    var level = value * 90;
+
+    if (engine.getValue("[Channel1]", "pfl") || engine.getValue("[Channel2]", "pfl")) {
+        if (group === "[Channel1]") {
+            midi.sendShortMsg(0xB0, 0x1F, level);
+        } else if (group === "[Channel2]") {
+            midi.sendShortMsg(0xB1, 0x1F, level);
+        }
+    } else if (group === "[Channel1]") {
+        midi.sendShortMsg(0xB0, 0x1F, level);
+    } else if (group === "[Channel2]") {
+        midi.sendShortMsg(0xB1, 0x1F, level);
+    }
+};
+
+MixtrackProFX.scratchToggle = function(channel) {
+    MixtrackProFX.scratchModeEnabled[channel] = !MixtrackProFX.scratchModeEnabled[channel];
+    midi.sendShortMsg(0x90 | channel, 0x07, MixtrackProFX.scratchModeEnabled[channel] ? 0x7F : 0x01);
+};
+
+// below: choose scratch A OR scratch B
+
+// ==== SCRATCH A START ==== //
+
+MixtrackProFX.wheelTouch = function(channel, control, value) {
+    var deckNumber = channel + 1;
+
+    if (!MixtrackProFX.shifted && MixtrackProFX.scratchModeEnabled[channel] && value === 0x7F) {
+        // touch start
+
+        engine.scratchEnable(deckNumber, MixtrackProFX.jogScratchSensitivity, 33+1/3, MixtrackProFX.jogScratchAlpha, MixtrackProFX.jogScratchBeta, true);
+    } else if (value === 0) {
+        // touch end
+        engine.scratchDisable(deckNumber, true);
+    }
+};
+
+MixtrackProFX.wheelTurn = function(channel, control, value, status, group) {
+    var deckNumber = channel + 1;
+
+    var newValue = value;
+
+    if (value >= 64) {
+        // correct the value if going backwards
+        newValue -= 128;
+    }
+
+    if (MixtrackProFX.shifted) {
+        // seek
+        var oldPos = engine.getValue(group, "playposition");
+
+        if (oldPos <= 0) {
+            oldPos = 0;
+        }
+
+        if (oldPos >= 1) {
+            oldPos = 1;
+        }
+
+        engine.setValue(group, "playposition", oldPos + newValue / MixtrackProFX.jogSeekSensitivity);
+    } else if (MixtrackProFX.scratchModeEnabled[channel] && engine.isScratching(deckNumber)) {
+        // scratch
+        engine.scratchTick(deckNumber, newValue);
+    } else {
+        // pitch bend
+        engine.setValue(group, "jog", newValue / MixtrackProFX.jogPitchSensitivity);
+    }
+};
+
+// ==== SCRATCH A END ==== //
```

Comment:
The following commented out code needs to be either deleted or enabled. No dead code.

Reply:
I removed the unused code. The reason why this existed in the first place was that one user on the forum [reported](https://mixxx.discourse.group/t/numark-mixtrack-pro-fx/19561/38) that scratching is unresponsive, and suggested replacing the jogwheel logic with the original code from @bad1dea5 original mapping. I decided to add it as an alternate mapping ("scratch B"). For me the simpler jogwheel logic ("scratch A") feels better, so I left it.